src/enum_net.cpp

```diff
@@ -377,6 +377,38 @@ namespace libtorrent
 		return false;
 	}
 
+#ifdef TORRENT_WINDOWS
+	HMODULE GetIPHelperHandle()
+	{
+		static bool handle_checked = false;
+		static HMODULE handle = 0;
+		if (!handle_checked)
+		{
+			handle = LoadLibraryA("Iphlpapi.dll");
+			handle_checked = true;
+		}
+		return handle;
+	}
+
+	template <typename T>
+	bool GetIPHelperProc(LPCSTR name, T& proc_out)
```

Comment:
I have this feeling that it might be nicer to return the function pointer, rather than taking an out-parameter. The type wouldn't be deducible anymore, but maybe that wouldn't be that big of a deal. it could be used like this:

auto GetAdaptersAddresses = get_ip_helper_proc<GetAdaptersAddresses_t>("GetAdapterAddresses");

if it fails, it could return ``nullptr``